tilavarauspalvelu/models/user/queryset.py

```diff
@@ -21,12 +21,37 @@ def remove_old_superuser_and_staff_permissions(self) -> None:
 
         self.filter(
             models.Q(is_active=True)
-            & models.Q(last_login__date__lt=cutoff)
             & (
                 models.Q(is_staff=True)  #
                 | models.Q(is_superuser=True)
             )
+            & models.Q(last_login__date__lt=cutoff)
         ).update(is_staff=False, is_superuser=False)
 
+    def should_deactivate_permissions(self) -> models.QuerySet:
```

Comment:
Wasn't explicitly specified in the ticket, but if the task is run daily then several emails would be sent to the same user, until either the permissions have eventually been deactivated or the user has logged in again -- and for the period of time (in production environment) between email and deactivation, the amount would be a bit excessive.
I guess need to store on the user model that this notification has been sent, and clear it on login.

Reply:
Added field `sent_email_about_deactivating_permissions` to user and then checks for this field when gathering email recipients for the email, as well as an update to the after the emails have been sent.